keras_cv/layers/preprocessing/random_rotation.py

```diff
@@ -226,6 +233,28 @@ def augment_bounding_boxes(
     def augment_label(self, label, transformation, **kwargs):
         return label
 
+    def augment_segmentation_mask(self, segmentation_mask, transformation, **kwargs):
+        # If segmentation_classes is specified, we have a dense segmentation mask.
+        # We therefore one-hot encode before rotation to avoid bad interpolation
+        # during the rotation transformation. Later, we make the mask sparse
+        # again using tf.argmax.
+        if self.segmentation_classes:
+            one_hot_mask = tf.one_hot(
+                tf.squeeze(segmentation_mask, axis=-1), self.segmentation_classes
+            )
+        else:
+            one_hot_mask = segmentation_mask
+
+        rotated_one_hot_mask = self._rotate_image(one_hot_mask, transformation)
+
+        if self.segmentation_classes:
```

Comment:
readability wise, I think it will be more clear to have:
```
if self.segmentation_classes:
  # change to one hot
  # rotate the one hot
  # argmax
  # return result
else:
  # rotate mask
  # return tf.round(result)

Reply:
Agreed -- done.